pkg/plugins/runtime/k8s/controllers/gatewayapi/converter.go

```diff
@@ -0,0 +1,175 @@
+package controllers
+
+import (
+	"errors"
+	"fmt"
+
+	"google.golang.org/protobuf/types/known/wrapperspb"
+	gatewayapi "sigs.k8s.io/gateway-api/apis/v1alpha2"
+
+	mesh_proto "github.com/kumahq/kuma/api/mesh/v1alpha1"
+)
+
+func serviceTagForGateway(name, namespace string) string {
+	return fmt.Sprintf("%s_%s_gateway", name, namespace)
+}
+
+func k8sToKumaHeader(header gatewayapi.HTTPHeader) *mesh_proto.GatewayRoute_HttpRoute_Filter_RequestHeader_Header {
+	return &mesh_proto.GatewayRoute_HttpRoute_Filter_RequestHeader_Header{
+		Name:  string(header.Name),
+		Value: header.Value,
+	}
+}
+
+func gapiToKumaRef(objectNamespace string, ref gatewayapi.BackendObjectReference) (map[string]string, error) {
+	// References to Services are required by GAPI to include a port
+	if ref.Port == nil {
+		return nil, errors.New("backend reference must include port")
+	}
+
+	namespace := objectNamespace
+	if ref.Namespace != nil {
+		namespace = string(*ref.Namespace)
+	}
+
+	return map[string]string{
+		mesh_proto.ServiceTag: fmt.Sprintf("%s_%s_svc_%d", ref.Name, namespace, *ref.Port),
+	}, nil
+}
+
+func gapiToKumaMatch(match gatewayapi.HTTPRouteMatch) *mesh_proto.GatewayRoute_HttpRoute_Match {
+	kumaMatch := &mesh_proto.GatewayRoute_HttpRoute_Match{}
+
+	if m := match.Method; m != nil {
+		if m, ok := mesh_proto.HttpMethod_value[string(*m)]; ok {
+			kumaMatch.Method = mesh_proto.HttpMethod(m)
+		}
+	}
+
+	if p := match.Path; p != nil {
+		path := &mesh_proto.GatewayRoute_HttpRoute_Match_Path{}
+		if v := p.Value; v != nil {
+			path.Value = *v
+		}
```

Comment:
p.Value is defaulted, so it should never be empty. If it is empty, that is an error case.

Reply:
Sounds good, not checking it at all then.